packages/flutter/test/material/drop_down_test.dart

```diff
@@ -181,4 +183,106 @@ void main() {
     await tester.pump();
     await tester.pump(const Duration(seconds: 1)); // finish the menu animation
   });
+
+  testWidgets('Drop down button aligns selected menu item', (WidgetTester tester) async {
+    Key buttonKey = new UniqueKey();
+    List<String> items = <String>['one', 'two', 'wide three', 'four'];
+    String value = 'two';
+
+    Widget build() {
+      return new MaterialApp(
+        home: new Material(
+          child: new Center(
+            child: new DropdownButton<String>(
+              key: buttonKey,
+              value: value,
+              onChanged: (String value) { },
+              items: items.map((String item) {
+                return new DropdownMenuItem<String>(
+                  key: new ValueKey<String>(item),
+                  value: item,
+                  child: new Text(item),
+                );
+              }).toList(),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(build());
+    RenderBox buttonBox = tester.renderObject(find.byKey(buttonKey));
+    assert(buttonBox.attached);
+
+    await tester.tap(find.text('two'));
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1)); // finish the menu animation
+
+    // The selected dropdown item is both in menu we just popped up, and in
+    // the IndexedStack contained by the dropdown button. Both of them should
+    // have the same origin and height as the dropdown button.
+    List<RenderObject> itemBoxes = tester.renderObjectList(find.byKey(new ValueKey<String>('two'))).toList();
```

Comment:
Maybe in fact use the text selection API to get the exact positions of the text boxes and compare those.